src/components/FieldsetItems/FieldsetItems.tsx

```diff
@@ -24,12 +24,23 @@ export type FieldsetItemsProps = {
  *
  * Helper sub-component for styling the control elements in the component.
  */
-export const FieldsetItems = ({
+export const FieldsetItems = <T extends ElementType = 'div'>({
   children,
-  as: Component = 'div',
+  as,
   className,
-}: FieldsetItemsProps) => {
+  ...props
+}: FieldsetItemsProps<T> &
+  Omit<ComponentPropsWithoutRef<T>, keyof FieldsetItemsProps<T>>) => {
```

Comment:
Here we are saying to use `"div"` as the default element type, and the params are our custom props and React's default props minus our custom ones (to avoid duplicates with any ones we redefine like `className`)

Reply:
Thanks for investigating this and figuring it out! I was about to say we can punt on it and leave the `//@ts-expect-error` if needbe